example/cpp-example/cronos.cc

```diff
@@ -493,6 +521,111 @@ void test_erc20_total_supply() {
   assert(total_supply == u256("100000000000000000000000000000000"));
 }
 
+// sample code for calling smart-contract
+void test_dynamic_api_send() {
+  std::ifstream t("../../common/src/contract/erc721-abi.json");
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+  std::string json = buffer.str();
+
+  String mymnemonics = getEnv("MYMNEMONICS");
+  String mycronosrpc = getEnv("MYCRONOSRPC");
+  String mycontract = getEnv("MYCONTRACT721");
+  int mychainid = stoi(getEnv("MYCRONOSCHAINID").c_str());
+  Box<Wallet> mywallet = createWallet(mymnemonics);
+
+  String senderAddress = mywallet->get_eth_address(0);
+  String receiverAddress = mywallet->get_eth_address(2);
+  auto thisNonce = get_eth_nonce(senderAddress.c_str(), mycronosrpc);
+  std::string tokenid;
+  std::cout << "Enter tokenid: ";
+  std::cin >> tokenid;
+
+  Box<EthContract> w = new_eth_contract(json);
+  w->add_address(senderAddress);                     // from
+  w->add_address(receiverAddress);                   // to
+  w->add_uint(tokenid);                              // tokenId
+  Vec<uint8_t> data = w->encode("safeTransferFrom"); // encoded
+  std::string hexstring = bytes_to_hexstring((char *)data.data(), data.size());
+  char hdpath[100];
+  int cointype = 60;
+  int chainid = mychainid; // defined in cronos-devnet.yaml
+  snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/0", cointype);
+  Box<PrivateKey> privatekey = mywallet->get_key(hdpath);
+
+  EthTxInfoRaw eth_tx_info = new_eth_tx_info();
+  eth_tx_info.to_address = mycontract;
+  eth_tx_info.nonce = thisNonce;
+  eth_tx_info.amount = "0";
+  eth_tx_info.amount_unit = EthAmount::EthDecimal;
+  eth_tx_info.data = data;
+  eth_tx_info.gas_limit = "219400";
+  eth_tx_info.gas_price = "100000000";
+  eth_tx_info.gas_price_unit = EthAmount::WeiDecimal;
+
+  Vec<uint8_t> signedtx =
+      build_eth_signed_tx(eth_tx_info, chainid, true, *privatekey);
+  String status =
+      broadcast_eth_signed_raw_tx(signedtx, mycronosrpc, 1000).status;
+  cout << "status: " << status << endl;
```

Comment:
with `DynamicContract` + String API, it can be something like:
```cpp
DynamicContractWrapper contract = new_dynamic_contract(mycontract, json,
                             mycronosrpc, chainid);
std::stringstream buffer;
buffer << "[" << senderAddress << ", " << receiverAddress << ", " << tokenid << "]";
std::string params = buffer.str();
String status = contract
               .send("safeTransferFrom", params,
                                   *privatekey)
               .status;
```

Reply:
ok, i'll check

Reply:
you may need to create some extra helper similar to `broadcast_contract_transfer_tx_blocking` etc. or @damoncro can advise whether any of the existing contract broadcast/query functions could be extended / refactored to allow for dynamic contracts

Reply:
Let me check~

Reply:
Refactoring the existing ERC20/ERC721/ERC1155 functions are possible. It could make all functions united and ruled under Dynamic Contracts. But I guess no big benefits, as those functions are standard functions.

But refactoring is still good because it could extending the missing extension functions, which we are missing and untested, for example enumeration on ERC721.